RecoTracker/TkDetLayers/src/TIDLayer.cc

```diff
@@ -171,7 +171,7 @@ void TIDLayer::groupedCompatibleDetsV(const TrajectoryStateOnSurface& startingSt
 #else
   constexpr int ringOrder[3]{1, 2, 0};
 #endif
-  auto index = [&ringIndices, &ringOrder](int i) { return ringOrder[ringIndices[i]]; };
+  auto index = [&ringIndices](int i) { return ringOrder[ringIndices[i]]; };
```

Comment:
Quick test on godbolt
* gcc 9 and clang 8 require `&ringOrder` in the capture
* icc >= 17 and clang 9 work without `&ringOrder` in the capture
  * I tested icc because of the `#ifdef __INTEL_COMPILER` above, but with `constexpr` (so the `#ifdef` is no longer needed in fact)

Reply:
@makortel indeed I had not tried the standard build, in this case anyway if we want to get rid of the warning we need some ifdef for the auto declaration based on the clang version, provided this can be accessible. Or do you have an alternative suggestion?
@smuzaffar I do see hw to test whether clang is in use, but not its version, do we have an handle for that already?

Reply:
according to https://stackoverflow.com/questions/1617877/how-to-detect-llvm-and-its-version-through-define-directives it should be just ```__clang__major__```

Reply:
> Or do you have an alternative suggestion?

On the to of my head no, but some googling pointed to https://stackoverflow.com/questions/55136414/constexpr-variable-captured-inside-lambda-loses-its-constexpr-ness and indeed `static constexpr int ringOrder[3]{1, 2, 0};` seems to work without capturing `&ringOder` for all gcc >= 5, clang >= 5, and icc >= 17.

Reply:
@makortel ok, let me try, it would be more elegant that what I have just committed (compiling in both gcc8 and CLANG 9)